dbus/methods_test.go

```diff
@@ -130,6 +133,238 @@ func TestStartStopUnit(t *testing.T) {
 	}
 }
 
+// Ensure that ListUnitsByNames works.
+func TestListUnitsByNames(t *testing.T) {
+	target1 := "systemd-journald.service"
+	target2 := "unexisting.service"
+
+	conn := setupConn(t)
+
+	units, err := conn.ListUnitsByNames([]string{target1, target2})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var unit *UnitStatus
+	for _, u := range units {
+		if u.Name == target1 {
+			unit = &u
+			break
+		}
+	}
+
+	if unit == nil {
+		t.Fatalf("%s unit not found in list", target1)
+	}
+
+	if unit.ActiveState != "active" {
+		t.Fatalf("%s unit should be active but it is %s", target1, unit.ActiveState)
+	}
+
+	unit = nil
+	for _, u := range units {
+		if u.Name == target2 {
+			unit = &u
+			break
+		}
+	}
+
+	if unit == nil {
```

Comment:
@jonboulle yep I confirm it's a bit confusing, but that's how systemd works actually please check this comment https://github.com/systemd/systemd/pull/3182#issuecomment-216828850 so basically the implementation doesn't bother it's just gives you back what you have requested setting up the right state of course.

Now this test can be updated to be more readable:
just create a target1 and target2 object which has {unit_name, activestate, a bolean found or not} , do the whole check inside the same loop, and check after the loop if the units have been found or not.  The doc of ListUnitsByNames() clearly states that it may return results for even non-existent units which is good.